File: webapp/src/components/Post.js
import "./../styles/Post.css"

export default function Post(props) {
    return (
        <>
            <div>
                <h2 className="post__title">{props.title}</h2>
                <small>{convertDateTimeOffset(props.created)}</small>
                <p>{props.summary}</p>
                <small>View count: {props.viewCount}</small>
                <small>Upvote: {props.upvote}</small>
                <small>Downvote: {props.downvote}</small>
            </div>
        </>
    );
}

function convertDateTimeOffset(dateTimeOffset) {
    let date = new Date(dateTimeOffset);
    return `${date.getHours()}:${date.getMinutes()} ${date.getDate()}/${date.getMonth() + 1}/${date.getFullYear()}`;
}

File: services/Blog/Domain/Post.cs
namespace Domain
{
    public class Post : AuditableEntity
    {
        public int Id { get; set; }

        private string _title = string.Empty;

        public string Title
        {
            get { return _title; }
            set
            {
                if (string.IsNullOrWhiteSpace(value))
                {
                    throw new ArgumentException("Title cannot be null.");
                }
                _title = value;
            }
        }

        private string _content = string.Empty;

        public string Content
        {
            get { return _content; }
            set
            {
                if (string.IsNullOrWhiteSpace(value))
                {
                    throw new ArgumentException("Content cannot be null.");
                }
                _content = value;
            }
        }

        private string _summary = string.Empty;

        public string Summary
        {
            get { return _summary; }
            set
            {
                if (string.IsNullOrWhiteSpace(value))
                {
                    throw new ArgumentException("Summary cannot be null.");
                }
                _summary = value;
            }
        }

        public int ViewCount { get; set; }
        public int Upvote { get; set; }
        public int Downvote { get; set; }
    }
}